refactor!: Context APIs changes and documentation/onboarding (#180)
spotify/confidence-sdk-swift: README.md

- The `loggerLevel` sets the verbosity level for logging to console. This can be useful while testing your integration with the Confidence SDK.
- `withContext()` sets the initial context. The context is a key-value map used for sampling and for targeting, so it determines how flags are evaluated by the Confidence backend.

_Note: the Confidence SDK has been intended to work as a single instance in your Application.
Creating multiple instances in the same runtime could lead to unexpected behaviours._

### Initialization strategy

After creating the confidence instance, you can choose between different strategies to initialize the SDK:
- `await confidence.fetchAndActivate()`: async function that fetches the flags from the Confidence backend according to the current context,
stores the result in storage, and make the same data ready for the Application to be consumed.

- `confidence.activate()`: this loads previously fetched flags data
from storage and makes that available for the Application to consume right away.

If you wish to avoid waiting on backend calls when the Application starts, the suggested approach is to call

> `confidence.activate()` and then call `confidence.asyncFetch()` to update the flag values in storage to be used on a future `activate()`. However, note that `activate()` ignores the current context: even if the current context has changed since the last fetch, flag values from the last fetch will be exposed to the Application.

Reviewer: I realised when reading this (combined with thinking about `app version` as a context value) that "multi context storage" would probably make us return default values after an app version update. 

Future thinking work: Should we have support for parts of context that do not affect context storage location?

Author: I think having different parts of contexts behaving differently according to their importance/use-case could enable better integrations, definitely something to look into in the future

### Managing the context
The context is set when instantiating the Confidence instance, but it can be updated at runtime:

```swift
await confidence.putContext(context: ["key": ConfidenceValue(string: "value")])

> await confidence.putContext(key: "key", value: ConfidenceValue(string: "value"))

Reviewer: It is a little unclear to me how these two functions differ.
`putContext(key: "key", value: ConfidenceValue(string: "value"))` is clear that it appends or changes a value.
I would be worried that calling `confidence.putContext(context: ["key": ConfidenceValue(string: "value")])` would reset my whole context.

Author: I agree. Let's come up with better names and ensure they are the same across Android and iOS SDKs

await confidence.removeContext(key: "key")
```

These functions are async functions, because the flag values are fetched from the backend for the new context, put in storage and then exposed to the Application.

> _Note: any change of the context in the Application could cause a change in the flag values._

Reviewer: ```suggestion
_Note: Changing the context in the Application could cause a change in the flag values._
```
